scheduler/crates/infra/src/repos/calendar/mod.rs

```diff
@@ -52,12 +52,11 @@ mod tests {
         ctx.repos.accounts.insert(&account).await.unwrap();
         let user = User::new(account.id.clone());
         ctx.repos.users.insert(&user).await.unwrap();
-        let mut calendar = Calendar::new(&user.id, &account.id);
+        let calendar = Calendar::new(&user.id, &account.id);
 
         // Insert
         assert!(ctx.repos.calendars.insert(&calendar).await.is_ok());
-
-        calendar.settings.week_start += 1;
+        calendar.settings.week_start.succ();
```

Comment:
This test seems to not actually test `save` method. `calendar.settings.week_start.succ();` seems to not increase `week_start`  but only return the next `Weekday`. So the calendar that is saved is exactly the same as the one inserted and the test will pass even if save method does not update the values.

Reply:
Oh yes, nice catch. Will fix it now.